### PR TITLE
Removed `--use-fake-symbol-table` option.

### LINKED ISSUE #13399: stats: Remove --use-fake-symbol-tables option.
See also https://github.com/envoyproxy/envoy/issues/6307 which removes the code supporting fake symbol tables, but leaves the option in place.

Removing the option is a breaking change, so splitting that off on its own.

### PR DESCRIPTION
**Pull request by mk46**
Signed-off-by: Manish Kumar <manish.kumar1@india.nec.com>
 
Commit Message: Removed `--use-fake-symbol-table` option.
Additional Description: Removing `--use-fake-symbol-table` because code for supporting the fake symbol table had been removed by #6307 
Risk Level:
Testing: unit and format test
Docs Changes: admin.rst
Release Notes: done
Fixes #13399 
Deprecated: `--use-fake-symbol-table` option

### CODE REVIEW

**Comment by mk46** [comment]
@jmarantz  Would you mind to take a look? Thanks!

**Review by jmarantz — COMMENTED** [review]
basically looks good, with one opportunity for further cleanup or leave the TODO in place (your choice).

File: include/envoy/stats/symbol_table.h [hunk 1]
```diff
@@ -138,16 +138,7 @@ class SymbolTable {
...
    * Calls the provided function with a string-view representation of the
-   * elaborated name. This is useful during the interim period when we
-   * are using FakeSymbolTableImpl, to avoid an extra allocation. Once
-   * we migrate to using SymbolTableImpl, this interface will no longer
-   * be helpful and can be removed. The reason it's useful now is that
-   * it makes up, in part, for some extra runtime overhead that is spent
-   * on the SymbolTable abstraction and API, without getting full benefit
-   * from the improved representation.
-   *
-   * TODO(#6307): Remove this when the transition from FakeSymbolTableImpl to
```
[the remark this answers is not in the document]
Comment (mk46) [reply include/envoy/stats/symbol_table.h:149]: Thanks for the review. I had removed the call-site for callWithStringView.

**Review by jmarantz — COMMENTED** [review]
There's one more slight edit needed in https://github.com/envoyproxy/envoy/blob/master/source/docs/stats.md to just delete the paragraph:
> As of October 6, 2020, the "fake" symbol table implementation has been removed from the system, and the "--use-fake-symbol-table" option is now a no-op, triggering a warning if set to "1". The option will be removed in a later release.

File: include/envoy/stats/symbol_table.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmarantz) [reply include/envoy/stats/symbol_table.h:149]: perfect!

**Review by jmarantz — APPROVED** [review]
look great modulo last stats.md edit and @envoyproxy/senior-maintainers approval.

**Comment by mk46** [comment]
> There's one more slight edit needed in https://github.com/envoyproxy/envoy/blob/master/source/docs/stats.md to just delete the paragraph:
> 
> > As of October 6, 2020, the "fake" symbol table implementation has been removed from the system, and the "--use-fake-symbol-table" option is now a no-op, triggering a warning if set to "1". The option will be removed in a later release.

Done.

**Review by mattklein123 — APPROVED** [review]
Awesome!

**Pull request merged**